auth.c

```diff
@@ -18,23 +18,34 @@
 #include "auth.h"
 #include "debug.h"
 #include "mutex.h"
+#include "utils.h"
+
+#include <string.h>
+#include <openssl/hmac.h>
```

Comment:
As a code convention, system and library includes go before our own local includes.